src/utils/commands/command_util.py

```diff
@@ -109,6 +109,25 @@ def rank(self, discord_id: str) -> str:
             )
         return result
 
+    def completion_check(self, question: dict, leetcode_user_id: str) -> int:
+        """
+        Helper function for status & group_status to check for question completion
+        """
+        completed = 0
+        question["complete"] = self.database.question_completions.check_completion(
+            leetcode_user_id, question["title_slug"]
+            )
+        if question["complete"]:
+            completed += 1
+        else:
+            question["complete"] = self.leetcode.check_challenge_completion(
+                leetcode_user_id, question["title_slug"]
+            )
+            if question["complete"]:
+                completed += 1
+                self.database.question_completions.insert(leetcode_user_id, question["title_slug"])
```

Comment:
A method to check completion should not also conditionally make database insertions. We should insert into the database outside of this function.